### DISCUSSION TITLE
Make YARP follow redirects

### DISCUSSION THREAD

**Discussion by nixtar** [dotnet/yarp#2113] [Q&A] [answered]
Hello,

I have a scenario where I need YARP to follow redirects (http 302) instead of returning them to the client.
The content is static HTTP content and doesn't require any tricky cookies or auth handling.

I've tried setting `AllowAutoRedirect` to true via the following:

```c#
builder.Services.AddReverseProxy()
    .ConfigureHttpClient((context, handler) =>
    {
        handler.AllowAutoRedirect = true;
    })
    .LoadFromMemory(GetRoutes(), GetClusters());
```

My routes and clusters are very basic catch all to a single http endpoint.

Is there any way to configure YARP to follow the redirection?

**Comment by nixtar** [comment]
I was able to get a vertical slice of the app working using a HttpForwarder instead of the full Reverse Proxy.

```c#
var builder = WebApplication.CreateBuilder(args);
builder.Services.AddHttpForwarder();

var app = builder.Build();

var httpClient = new HttpMessageInvoker(new SocketsHttpHandler()
{
    UseProxy = false,
    AllowAutoRedirect = true,
    ConnectTimeout = TimeSpan.FromSeconds(30)
});

var transformer = HttpTransformer.Empty;
var requestOptions = new ForwarderRequestConfig();

app.MapForwarder("/{**catch-all}", "http://endpointhostname", requestOptions, transformer, httpClient);

app.Run();
```

Perhaps a `IForwarderHttpClientFactory` can have the same result?

Out of time for today, will update this thread if I make any further progress.

**Reply by Tratcher** [reply, marked as answer]
Your first example should have worked, it's doing essentially the same thing.

**Reply by nixtar** [reply]
Thanks for the reply, that's what I expected to work also. 🤔 Good to know I'm not entirely crazy. 😋

I'll make a new project that reproduces the settings I had in my first attempt, if it's still not working as intended I'll log an issue.
